src/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -748,20 +752,92 @@ const UsersConfigurationRow = ({
     );
 };
 
+const validateKey = key => cockpit.spawn(["ssh-keygen", "-l", "-f", "-"], { error: "message" }).input(key);
+
+// This method needs to be outside of component as re-render would create a new instance of debounce
+const parseKey = debounce(500, (key, setKeyObject) => {
+    if (isEmpty(key))
+        return;
+
+    return validateKey(key)
+            .then(() => {
+                const parts = key.split(" ");
+                if (parts.length > 2) {
+                    setKeyObject({
+                        type: parts[0],
+                        data: parts[1],
+                        comment: parts[2], // comment is optional in SSH-format
+                    });
+                }
+            })
+            .catch(() => {
+                setKeyObject(undefined);
+                console.warn("Could not validate the public key");
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1150-20231004-083449-642f6dc5-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)